src/main/java/com/google/sps/data/CalendarClientData.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.*;
+
+/** Class containing the user's free hours. */
+public final class CalendarClientData {
```

Comment:
This class is directly serialized into the response using GSON, which serializes all private fields. The *private* field types/names of this class are part of the *public* API of your servlet. Therefore, I think it should be called `CalendarDataResponse` and contain only fields, constructor, and getters.

Move the logic into a separate utility. (Keep the initialize + addEvent pattern - I like it!) You can either have a method on the utility called getCalendarDataResponse, or just create the response in your servlet.

Reply:
`Resolved`